spec/annotations/semantics.yaml

```diff
@@ -36,112 +69,207 @@ definitions:
   "@Semantics.calendar.dayOfMonth":
     <<: *semanticsFieldBoolean
     description: |-
+      The element states the day of month.
+    x-extension-targets:
+      - IntegerType
+      - StringType
```

Comment:
I added extension-targets (scope) to the other extensions as well. Please check if they are correct.

With dates I was not sure if they are always Integer, or also allowed for String. If it's string, similar question as above. With month string could also be interpreted as something like "october", "Oktober", etc. ??